From b3ee03b155dad86bad8fd197764e2d340b1eb8c6 Mon Sep 17 00:00:00 2001
From: Maxim Polyakov <m.polyakov@yadro.com>
Date: Mon, 8 Jul 2019 12:13:18 +0300
Subject: [PATCH 03/18] discover/platform-powerpc: return the actual mailbox
 size

get_ipmi_boot_mailbox_block() should return the actual size
of the received IPMI mailbox data

Signed-off-by: Maxim Polyakov <m.polyakov@yadro.com>
(cherry picked from commit 5f8321eb6479113a09c771c49e6781571ec8e2e2)
Signed-off-by: Klaus Heinrich Kiwi <klaus@linux.vnet.ibm.com>
---
 discover/platform-powerpc.c | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/discover/platform-powerpc.c b/discover/platform-powerpc.c
index 1e33bf1..c874560 100644
--- a/discover/platform-powerpc.c
+++ b/discover/platform-powerpc.c
@@ -473,12 +473,7 @@ static int get_ipmi_boot_mailbox_block(struct platform_powerpc *platform,
 		return -1;
 	}
 
-	if (resp_len == 4) {
-		pb_debug_fn("block %hu empty\n", block);
-		return 0;
-	}
-
-	blocksize = sizeof(resp) - 4;
+	blocksize = resp_len - 4;
 	pb_debug_fn("Mailbox block %hu returns only %zu bytes in block\n",
 			block, blocksize);
 
@@ -512,6 +507,12 @@ static int get_ipmi_boot_mailbox_block(struct platform_powerpc *platform,
 		return -1;
 	}
 
+	if (!blocksize) {
+		pb_debug_fn("block %hu empty\n", block);
+		return 0;
+	}
+
+
 	memcpy(buf, &resp[4], blocksize);
 
 	return blocksize;
-- 
2.17.1

